e2e/aws/rds_test.go

```diff
@@ -0,0 +1,224 @@
+		// itself IAM rds:connect permissions.
+		require.EventuallyWithT(t, func(c *assert.CollectT) {
+			var err error
+			pgConn, err = pgconn.Connect(ctx, connString)
+			assert.NoError(c, err)
+			assert.NotNil(c, pgConn)
+		}, time.Second*10, time.Second, "connecting to postgres")
+
+		// Execute a query.
+		results, err := pgConn.Exec(ctx, "select 1").ReadAll()
+		require.NoError(t, err)
+		for i, r := range results {
+			require.NoError(t, r.Err, "error in result %v", i)
+		}
+
+		// Disconnect.
+		err = pgConn.Close(ctx)
+		require.NoError(t, err)
+	}
+}
+
+// startLocalALPNProxy starts local ALPN proxy for the specified database.
+func startLocalALPNProxy(t *testing.T, ctx context.Context, cluster *helpers.TeleInstance, route tlsca.RouteToDatabase) *alpnproxy.LocalProxy {
+	t.Helper()
+	proto, err := alpncommon.ToALPNProtocol(route.Protocol)
+	require.NoError(t, err)
+
+	listener, err := net.Listen("tcp", "127.0.0.1:0")
+	require.NoError(t, err)
+
+	proxyNetAddr, err := cluster.Process.ProxyWebAddr()
+	require.NoError(t, err)
+
+	authSrv := cluster.Process.GetAuthServer()
+	tlsCert := generateClientDBCert(t, authSrv, username, route)
+
+	proxy, err := alpnproxy.NewLocalProxy(alpnproxy.LocalProxyConfig{
+		RemoteProxyAddr:    proxyNetAddr.String(),
+		Protocols:          []alpncommon.Protocol{proto},
+		InsecureSkipVerify: true,
+		Listener:           listener,
+		ParentContext:      ctx,
+		Certs:              []tls.Certificate{tlsCert},
+	})
+	require.NoError(t, err)
+
+	go proxy.Start(ctx)
+
+	return proxy
+}
+
+// generateClientDBCert creates a test db cert for the given user and database.
+func generateClientDBCert(t *testing.T, authSrv *auth.Server, user string, route tlsca.RouteToDatabase) tls.Certificate {
+	t.Helper()
+	key, err := client.GenerateRSAKey()
+	require.NoError(t, err)
+
+	clusterName, err := authSrv.GetClusterName()
+	require.NoError(t, err)
+
+	clientCert, err := authSrv.GenerateDatabaseTestCert(
+		auth.DatabaseTestCertRequest{
+			PublicKey:       key.MarshalSSHPublicKey(),
+			Cluster:         clusterName.GetClusterName(),
+			Username:        user,
+			RouteToDatabase: route,
+		})
+	require.NoError(t, err)
+
+	tlsCert, err := key.TLSCertificate(clientCert)
+	require.NoError(t, err)
+	return tlsCert
+}
+
+func waitForDatabases(t *testing.T, auth *service.TeleportProcess, wantNames ...string) {
+	wantSet := utils.StringsSet(wantNames)
+	require.EventuallyWithT(t, func(c *assert.CollectT) {
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		defer cancel()
+		databases, err := auth.GetAuthServer().GetDatabases(ctx)
+		require.NoError(t, err)
+
+		// map the registered "db" resource names.
+		seen := map[string]struct{}{}
+		for _, db := range databases {
+			seen[db.GetName()] = struct{}{}
+		}
+		assert.Empty(c, cmp.Diff(wantSet, seen))
+	}, 3*time.Minute, 500*time.Millisecond, "waiting for the discovery service to create databases")
```

Comment:
For something that runs each 1min, testing the tick each 500ms seems to much
```suggestion
	}, 3*time.Minute, 3*time.Second, "waiting for the discovery service to create databases")
```

Reply:
[093bec5](https://github.com/gravitational/teleport/pull/29755/commits/093bec5940eac9a36dcbbf98ae2344809e87bc93)